spec/models/profile_spec.rb

```diff
@@ -1,4 +1,35 @@
 require "rails_helper"
+require "open-uri"
 
 RSpec.describe Profile, type: :model do
+  describe "Associations" do
+
+    it "must belong to a user" do
+      profile = (build(:profile))
+      expect(profile.user).to be_valid
+    end
+    
+    it "has a profile picture" do
+      profile = (build(:profile))
+      file = URI.open("https://images.unsplash.com/photo-1613679074971-91fc27180061?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1887&q=80")
+      profile.photo.attach(io: file, filename: "avatar.jpg")
+      profile.save
+      expect(profile.photo.attached?).to be_truthy
+    end
+  end
+
+  describe "Validations" do
+    it "Has a valid linkedin url" do
+      expect { create(:profile, linkedin: 'Boris Johnson') }.to raise_error(ActiveRecord::RecordInvalid)
```

Comment:
nice! Only improvement would be to run against the expected error, e.g. raise_error("Validation failed: store can't be negative"); not sure what the quick way of doing this would be though, is there a way we can list all possible errors?

Reply:
here's a quick example I've just used in mine:

      context "title too long" do
        it "is not valid with too long a title" do
          subject.title = ("a" * 51).to_s
          expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid, "Validation failed: Title is too long (maximum is 50 characters)")
        end
      end